flink-connectors/flink-connector-jdbc/src/main/java/org/apache/flink/connector/jdbc/xa/JdbcXaSinkFunction.java

```diff
@@ -178,6 +178,8 @@ public JdbcXaSinkFunction(
                 new XaSinkStateHandlerImpl(),
                 options,
                 new XaGroupOpsImpl(xaFacade));
+        Preconditions.checkArgument(
+                executionOptions.getMaxRetries() == 0, "JDBC XA sink requires maxRetries = 0");
```

Comment:
1. With this check, it's still possible to construct `JdbcXaSinkFunction` with retries > 0 by passing it a pre-created `JdbcOutputFormat`.

WDYT about moving this check to the next constructor and validating `outputFormat.getExecutionOptions().getMaxRetries()` there? (new method)

2. Maybe it makes sense to clarify the reason in the message or refer to this issue?
3. And mention it in constructor javadoc?